src/main/scala/org/apache/spark/streamdm/classifiers/trees/HoeffdingTree.scala

```diff
@@ -232,35 +243,35 @@ class HoeffdingTreeModel(val espec: ExampleSpecification, val numericObserverTyp
     if (growthAllowed && learnNode.isInstanceOf[ActiveLearningNode]) {
       // split only happened when the tree is allowed to grow and the node is a ActiveLearningNode
       val activeNode = learnNode.asInstanceOf[ActiveLearningNode]
-      if (activeNode.addOnWeight() >= graceNum) {
-        val isPure = activeNode.isPure()
-        if (!isPure) {
-          // one best suggestion for each feature
-          var bestSuggestions: Array[FeatureSplit] = activeNode.getBestSplitSuggestions(splitCriterion, this)
-          //sort the suggestion based on the merit
-          bestSuggestions = bestSuggestions.sorted
-          if (shouldSplit(activeNode, bestSuggestions)) {
-            val best: FeatureSplit = bestSuggestions.last
-            if (best.conditionalTest == null) {
-              //deactive a learning node
-              deactiveLearningNode(activeNode, parent, pIndex)
-            } else {
-              logInfo("before Split:" + root.description())
-              //replace the ActiveLearningNode with a SplitNode and create children
-              val splitNode: SplitNode = new SplitNode(activeNode.classDistribution, best.conditionalTest)
-              for (index <- 0 until best.numSplit) {
-                splitNode.setChild(index,
-                  createLearningNode(learningNodeType, best.distributionFromSplit(index)))
-              }
-              // repalce the node
-              addSplitNode(splitNode, parent, pIndex)
-              logInfo("after Split:" + root.description())
+      val isPure = activeNode.isPure()
+      if (!isPure) {
+        // one best suggestion for each feature
+        var bestSuggestions: Array[FeatureSplit] = activeNode.getBestSplitSuggestions(splitCriterion, this)
+        //sort the suggestion based on the merit
+        bestSuggestions = bestSuggestions.sorted
+        if (shouldSplit(activeNode, bestSuggestions)) {
+          val best: FeatureSplit = bestSuggestions.last
+          if (best.conditionalTest == null) {
+            //deactivate a learning node
+            deactiveLearningNode(activeNode, parent, pIndex)
+          } else {
+            logInfo("split! ")
```

Comment:
This `logInfo` should be removed

Reply:
Do you think that it is good to keep the logInfo, so that the other contributors could easily debug the fundamental steps?

Reply:
It makes sense, let's keep it for now